Common/Securities/IDerivativeSecurityFilterUniverse.cs

```diff
@@ -14,14 +14,29 @@
  *
 */
 
+using System.Collections;
 using System.Collections.Generic;
 
 namespace QuantConnect.Securities
 {
+#pragma warning disable CA1010, CA1710 // Generic interface should also be implemented
+#pragma warning disable CA1710 // Identifiers should have correct suffix
+
+    /// <summary>
+    /// Represents derivative symbols universe used in filtering.
+    /// </summary>
+    public interface IDerivativeSecurityFilterUniverse : IEnumerable
```

Comment:
Are u sure we need this base interface don't think it's used, the simpler the merrier 🤔

Reply:
This was needed to simplify casting from derived filter universe clases to their interface in methods like `https://github.com/QuantConnect/Lean/blob/master/Common/Securities/FuncSecurityDerivativeFilter.cs#L37-L38`, since casting from base classes into base generic interfaces requires knowing the generic types.

Reply:
The interface was removed 👌